### ISSUE TITLE
Trouble specifying version ranges in Kobalt

### ISSUE THREAD

**Issue by pabl0rg** [cbeust/kobalt#318]
I checked out and tried http://maven.apache.org/components/enforcer/enforcer-rules/versionRanges.html

postgres has very weird versioning, so I'm not surprised this didn't work:
`compile("org.postgresql:postgresql:[9.4,)")`

```
Downloaded artifact org.postgresql:postgresql:pom:42.0.0-20170127.084917-23 from  (http://repository.jetbrains.com/all/, default, releases+snapshots)
*****
***** ERROR Error: com.beust.kobalt.KobaltException: Couldn't resolve org.postgresql:postgresql:jar:42.0.0-SNAPSHOT
```

the exception seems to have come from within Aether
```
org.eclipse.aether.transfer.ArtifactNotFoundException: Could not find artifact org.postgresql:postgresql:jar:42.0.0-20170127.084917-23 in  (https://jcenter.bintray.com/)
```

Note that, in gradle this did work (some people may care)
`compile group: 'org.postgresql', name: 'postgresql', version: '9.4.+'`

More surprising was that in Kobalt .917 this did not work:
`"org.sql2o:sql2o:[1.5,)"`

```
*****
***** ERROR Couldn't resolve range artifact org.sql2o:sql2o:jar:[1.5,)
*****
```

But this does:
`"org.sql2o:sql2o:1.5.4"`

**Comment by cbeust** [comment]
Some of it works:

```
$ ./kobaltw --resolve "org.postgresql:postgresql:[9.4,)"
║           /Users/cedricbeust/.kobalt/cache/org/postgresql/postgresql/9.4.1212.jre7/postgresql-9.4.1212.jre7.jar ║

$ ./kobaltw--resolve "org.postgresql:postgresql:[9.3,9.4)"
║           /Users/cedricbeust/.kobalt/cache/org/postgresql/postgresql/9.3-1104-jdbc41/postgresql-9.3-1104-jdbc41.jar ║
```

Note that the error message says "can't resolve **range** artifact, so Kobalt correctly identified it as such. Are you sure there exists an artifact that matches that version range `"org.sql2o:sql2o:jar:[1.5,)"`.

**Comment by pabl0rg** [comment]
well... it got the hard case (postgres)  but not the easy one. or i'm not understanding the range syntax.  The dependency does seem to exist

[https://mvnrepository.com/artifact/org.sql2o/sql2o/1.5.4](https://mvnrepository.com/artifact/org.sql2o/sql2o/1.5.4)

**Comment by cbeust** [comment]
Indeed. It should work.

I just learned that Aether is no longer supported, so that might explain why I'm beginning to find so many bugs in this implementation. I wouldn't be surprised if this was one more.

I'll have to investigate if I can instead use Sonatype's implementation of Aether and see if it helps, but that might be a lot of work.

**Comment by pabl0rg** [comment]
do you think it would be easier to use maven-resolver?  (it's what the maven folks seem to be using now)

https://git-wip-us.apache.org/repos/asf?p=maven.git;a=commitdiff;h=4d6c9292c433626fa0b9a1cf711d4d0c366e7f33

**Comment by cbeust** [comment]
Yes, I came across maven-resolver yesterday and I'm looking into transitioning to it.

**Comment by cbeust** [comment]
I actually got this to work with 0.916 simply by wiping my cache:

```
rm -rf ~/.kobalt/cache/org/sql2o
```

Can you try that?

```
$ k --resolve "org.sql2o:sql2o:[1.5.0,1.5.1]"
║           C:\Users\Cedric\.kobalt\cache\org\sql2o\sql2o\1.5.1\sql2o-1.5.1.jar ║
$ k --resolve "org.sql2o:sql2o:[1.5.0,1.5.)]"
║           C:\Users\Cedric\.kobalt\cache\org\sql2o\sql2o\1.5.1\sql2o-1.5.0.jar ║

```

**Comment by pabl0rg** [comment]
still failed for me in .916 or .918
but using version ranges may not even be a good idea

```
Juans-MacBook-Pro backend-1.7> rm -rf ~/.kobalt/cache/org/sql2o
Juans-MacBook-Pro backend-1.7> kbw --resolve "org.sql2o:sql2o:[1.5.0,1.5.1)"
            _  __          _               _   _
           | |/ /   ___   | |__     __ _  | | | |_
           | ' /   / _ \  | '_ \   / _` | | | | __|
           | . \  | (_) | | |_) | | (_| | | | | |_
           |_|\_\  \___/  |_.__/   \__,_| |_|  \__|  0.916

Done extracting: @val bs = buildScript {
    val r = repos("http://repository.jetbrains.com/all")
}

@
*****
***** ERROR Couldn't resolve range artifact org.sql2o:sql2o:jar:[1.5.0,1.5.1)
*****
```

**Comment by cbeust** [comment]
Yes, version ranges are a bad idea overall and you shouldn't use them for a bunch of different reasons.

Still, this should work, so it's bugging me :-)

Did you see any downloading happening after you wiped your cache? Is anything in your cache under `sql2o`?

The latest Kobalt uses maven-resolver but it should make no difference.

**Comment by cbeust** [comment]
Well, it seems to be erratic, now I can't get it to work either. Sometimes it does, other times it doesn't.

One more reason not to use version ranges I guess...

**Comment by cbeust** [comment]
This should be fixed in 0.921, can you try again?

**Comment by pabl0rg** [comment]
"full success!" thanks
